functions/Test-DbaLastBackup.ps1

```diff
@@ -393,7 +428,19 @@ Restores data and log files to alternative locations and only restores databases
 								}
 							}
 							
-							if ($destserver.Databases[$dbname] -ne $null -and !$NoDrop)
+							# Cleanup BackupFiles if -copyDestination and backup was moved to destination
+							if($copyDestination -eq $true -and $($lastbackup.path).startswith($($destserver.BackupDirectory)))
+							{
+								Write-Message -Level Verbose -Message "Removing backup file from $destination"
+								Remove-item $($lastbackup.fullname)
```

Comment:
In addition, I think what I'd consider doing is keeping an object filled with the new objects, and then just pipe that to Remove-Item if it accepts pipe. In this case, it would remove the last backup, not the new copy. I think?